### PR TITLE
Feature/rr 952 allow patch with no choices

### PR DESCRIPTION
**Pull request by au2**
If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### How should this be manually tested?

1) Create a choice or choices type question without any choices (empty array choices property).
2) Try to add choices using patch.

### CODE REVIEW

**Comment by Arghast** [comment]
@au2 @unprofessional Functionality looks good!

**Comment by unprofessional** [comment]
I'm out next week @jsachs and @orndorffgrant will be code reviewers with @Arghast as functional tester.

File: models/dao/question.dao.js [hunk 1]
```diff
@@ -452,7 +458,18 @@ module.exports = class QuestionDAO extends Translatable {
                         }
                         return null;
                     })
-                    .then(() => questions);
+                    .then(() => {
+                        if (scope !== 'summary') {
+                            questions.forEach((q) => {
+                                if (isDirectChoiceType(q)) {
```
Comment (jsachs) [inline models/dao/question.dao.js]: Let's combine these two `if`s.

Comment (au2) [reply models/dao/question.dao.js]: done

**Comment by au2** [comment]
@jsachs is this good to go now?

**Pull request merged**